e2e/cypress/integration/enterprise/system_console/download_compliance_export_file_spec.js

```diff
@@ -0,0 +1,202 @@
+        // Goto file storage settings Page
+        cy.visit('/admin_console/environment/file_storage');
+
+        // Get AWS credentials
+        const AWS_S3_BUCKET = Cypress.env('AWS_S3_BUCKET');
+        const AWS_ACCESS_KEY_ID = Cypress.env('AWS_ACCESS_KEY_ID');
+        const AWS_SECRET_ACCESS_KEY = Cypress.env('AWS_SECRET_ACCESS_KEY');
+
+        // Config AWS settings
+        cy.get('select[data-testid="FileSettings.DriverNamedropdown"]').select('amazons3');
+        cy.get('input[data-testid="FileSettings.AmazonS3Bucketinput"]').type(AWS_S3_BUCKET);
+        cy.get('input[data-testid="FileSettings.AmazonS3AccessKeyIdinput"]').type(AWS_ACCESS_KEY_ID);
+        cy.get('input[data-testid="FileSettings.AmazonS3SecretAccessKeyinput"]').type(AWS_SECRET_ACCESS_KEY);
+
+        // Save file storage settings
+        cy.get('button[data-testid="saveSetting"]').click();
+
+        waitUntilConfigSave();
+
+        // Goto compliance page and enable export
+        gotoCompliancePage();
+        enableComplianceExport();
+
+        // Navigate to a team and post an attachment
+        gotoTeamAndPostImage();
+
+        // Goto compliance page and start export
+        gotoCompliancePage();
+        exportCompliance();
+
+        // Get the download link
+        cy.get('@firstRow').findByText('Download').parents('a').should('exist').then((fileAttachment) => {
+            const fileURL = fileAttachment.attr('href');
+
+            // Download the file
+            downloadAttachmentAndVerifyItsProperties(fileURL);
+        });
+    });
+});
+
+function enableComplianceExport() {
+    // Enable compliance export
+    cy.get('input[data-testid="enableComplianceExporttrue"]').click();
+
+    // Change export format to CSV
+    cy.get('select[data-testid="exportFormatdropdown"]').select('csv');
+
+    // Save settings
+    cy.get('button[data-testid="saveSetting"]').click();
+
+    waitUntilConfigSave();
+}
+
+function gotoCompliancePage() {
+    cy.visit('/admin_console/compliance/export');
+}
+
+function gotoTeamAndPostImage() {
+    // Get user teams
+    cy.apiGetTeamsForUser().then(({teams}) => {
+        const team = teams[0];
+        cy.visit(`/${team.name}/channels/town-square`); // Go to the first found team
+    });
+    const file = {
+        filename: 'image-400x400.jpg',
+        originalSize: {width: 400, height: 400},
+        thumbnailSize: {width: 400, height: 400},
+    };
+    cy.get('#fileUploadInput').attachFile(file.filename);
+
+    // Wait until the image upload process start
+    cy.waitUntil(() => {
+        return Cypress.$('.post-image__uploadingTxt').length;
+    });
+
+    // Wait until the image is uploaded
+    cy.waitUntil(() => {
+        return !Cypress.$('.post-image__uploadingTxt').length;
+    }, {
+        timeout: TIMEOUTS.FIVE_MIN,
+        interval: TIMEOUTS.ONE_SEC,
+        errorMsg: 'Unable to upload attachment in time',
+    });
```

Comment:
This can be simplified to,

```suggestion
    // # Wait until the image is uploaded
    cy.waitUntil(() => cy.get('#postCreateFooter').then((el) => {
        return el.find('.post-image.normal').length > 0;
    }));
```